src/coprocessor/endpoint.rs

```diff
@@ -416,11 +418,13 @@ impl<E: Engine> Endpoint<E> {
 
         tracker.on_begin_all_items();
 
-        let handle_request_future = track(handler.handle_request(), &mut tracker);
+        let deadline = tracker.req_ctx.deadline;
+        let handle_request_future =
+            check_deadline(track(handler.handle_request(), &mut tracker), deadline);
         let result = if let Some(semaphore) = &semaphore {
-            limit_concurrency(handle_request_future, semaphore, LIGHT_TASK_THRESHOLD).await
+            limit_concurrency(handle_request_future, semaphore, LIGHT_TASK_THRESHOLD).await?
```

Comment:
Seems the logic is different from before.

Reply:
Do you mean the `?`?
As `ConcurrencyLimiter` does not return an error, the `?` handles the result returned by `DeadlineCheck`. So it does not look different to me

Reply:
I map the error inside `handle_unary_request_impl`